whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/BouncedMessageParser.java

```diff
@@ -45,35 +48,100 @@ public BouncedMessageParser(@Value("${mail.smtp.from:}") final String smtpFrom)
-            if (isReportDeliveryStatus(multipartReport)) {
-                final DeliveryStatus deliveryStatus = deliveryStatus(message);
-                if (isFailed(deliveryStatus)) {
-                    final MimeMessage returnedMessage = multipartReport.getReturnedMessage();
-                    final String messageId = getMessageId(returnedMessage.getMessageID());
-                    final List<String> recipient = extractRecipients(deliveryStatus);
-                    return new EmailMessageInfo(recipient, messageId);
+        if (isMultipartReport(message)) {
+            try {
+                final MultipartReport multipartReport = multipartReport(message.getContent());
+                if (isReportDeliveryStatus(multipartReport)) {
+                    final DeliveryStatus deliveryStatus = deliveryStatus(message);
+                    if (isFailed(deliveryStatus)) {
+                        final MimeMessage returnedMessage = getReturnedMessage(multipartReport);
+                        final String messageId = getMessageId(returnedMessage.getMessageID());
+                        final List<String> recipient = extractRecipients(deliveryStatus);
+                        return new EmailMessageInfo(recipient, messageId);
+                    }
+                }
+            } catch (MessagingException | IOException | IllegalStateException ex){
+                throw new MailParsingException("Error parsing multipart report");
+            }
+            // multipart report can *only* be a failure
+            throw new MailParsingException("MultiPart message without failure report");
+        }
+
+        if (isMultipartMixed(message)) {
+            try {
+                final MimeMultipart multipart = multipart(message.getContent());
+                if (isReportDeliveryStatus(multipart)) {
+                    final DeliveryStatus deliveryStatus = deliveryStatus(message);
+                    if (isFailed(deliveryStatus)) {
+                        final MimeMessage returnedMessage = getReturnedMessage(multipart);
+                        final String messageId = getMessageId(returnedMessage.getMessageID());
+                        final List<String> recipient = extractRecipients(deliveryStatus);
+                        return new EmailMessageInfo(recipient, messageId);
+                    }
                 }
+            } catch (MessagingException | IOException | IllegalStateException ex) {
+                LOGGER.error(String.format("%s: %s", ex.getClass().getName(), ex.getMessage()), ex);
+                throw new MailParsingException("Error parsing multipart report");
             }
-        } catch (MessagingException | IOException | IllegalStateException ex){
-            throw new MailParsingException("Error parsing multipart report");
         }
-        throw new MailParsingException("MultiPart message without failure report");
+
+        // fall through: message is not bounced message
+        return null;
```

Comment:
Maybe early check that it is a plain text and return null if it a plain text (at the beginning of the method?)
and return a MailParsingException here...this is more strict but we would just allow mail updates when text plain